[22기 최재상] 멘토님 리뷰 후 수정사항 반영

Higher77 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 멘토님이 리뷰해주신 코드들에 대한 수정을 하였습니다.
		 
## 수정 사항들 자세한 내용
FeedsData에 있던 잘못된 데이터 fetch를 삭제하였습니다.
분리되었던 Comments 데이터를 FeedsData에 병합하였습니다.
fetch함수의 위치를 Main으로 올렸습니다.

어울리지 않는 변수이름을 수정하였습니다.
변수이름을 지을 때 잘못사용한 네이밍 컨벤션을 수정하였습니다.

Main.scss의 main태그에 잘못된 height값이 사용된걸 삭제했습니다.
overflow:auto역시 삭제했습니다.

Main_right.scss의 .main_right에 height:400px로 고정값을 주었습니다.

## 기타 질문 및 특이 사항

1. 함수,데이터,변수들의 이름을 줄 때 지켜야할 컨벤션들이 헷갈립니다.




## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
[Remove] Main.scss 불필요한 body내용 삭제

Commit:
[Merge] 데이터 최신화를 위한 merge

Commit:
[Add]: Mission 1 - 사용자 입력 데이터 저장 기능 구현

Commit:
[Add] : Misson 2 - 로그인 버튼 활성화 기능 구현

Commit:
[Add] : Mission 3 - 댓글 기능 구현

Commit:
[Add] : Mission 4 - 댓글 컴포넌트화 , props로 데이터 전달 구현

Commit:
[Add] Mock data를 Main_right의 Recommendation_list에 적용

Commit:
[Add] feed에 mock데이터 적용, 댓글 달면 input value값 초기화 적용

Commit:
[Refactor] self-refactoring 진행

Commit:
[Add] 2차 Self-refactoring 작업 완료
-login.js의 handleIdInput함수와 handlePwInput을 handleInput으로 통합.

silviaoh commented on src/pages/jaesangChoi/Login/Login.js:
```diff
@@ -3,28 +3,57 @@ import './Login.scss';
         </div>
 
-        <Login_Button />
+        <Login_Button
+          idfunc={this.state.id.indexOf('@') === -1 ? 'failed' : 'passed'}
+          pwfunc={this.state.pw.length >= 5 ? 'nice' : 'shit'}
+          colorfunc={
+            this.state.id.indexOf('@') !== -1 && this.state.pw.length >= 5
+              ? 'blue'
+              : 'sky_blue'
+          }
```
this.state ... 이거 너무 기니까 이것도 비구조화 할당하시는 게 가독성이 좋을 것 같아요!

Higher77 replied:
말씀하신대로 바꿨습니다!

silviaoh commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -73,19 +132,28 @@ class Feeds extends Component {
             </div>
             <div className="push_comment_text">
               <input
+                onChange={this.handleComment}
+                onKeyPress={this.enterDown}
+                value={this.state.comment}
                 className="input_comment"
                 type="text"
                 placeholder="댓글달기..."
               />
             </div>
             <div className="push_comment_button">
-              <button>게시</button>
+              <button onClick={this.pressButton}>게시</button>
```
여기 form 태그로 감싸시고 onSubmit 이벤트주시면 엔터키 누를 때 실행되는 메소드 따로 안 만드셔도 pressButton 메소드 하나만으로 댓글 추가시킬 수 있어요!

Higher77 replied:
form에 이런 기능이 있는 줄 처음알았습니다! 덕분에 새로운 지식을 얻었습니다. 정말 감사해요!

Commit:
[Refactor] peer review - login.js <Login_Button>부분 비구조화 할당완료.

Commit:
[Refactor] feed.js 댓글 input 부모를 form으로 교체 후 enter함수 삭제
-input을 둘러싸고 있는 부모div를 form으로 교체하였습니다.
-form태그 안에 input태그가 '유일할 시'  엔터를 치게 되면 기본적으로 submit이 됩니다.
-form안의 onSubmit이벤트에 pressButton을 넣었습니다.
-하지만, 이 상황에서는 페이지가 새로고침되기 때문에  e.preventDefault로 기본 동작을 막았습니다.
-그래도 enter키를 누르면 submit는 작동합니다.

Commit:
[Add] 회원가입 및 로그인 구현 fetch함수 사용(백엔드와 소통)

Commit:
[remove] fetch적용하기 위해 주석처리한 함수들 제거

yeonuk-hwang commented:
재상님 3팀은 전반적으로 피어리뷰가 잘 안 이루어진 것 같은데 이유가 있을까요??

yeonuk-hwang commented on src/pages/jaesangChoi/Login/Login.js:
```diff
@@ -3,28 +3,56 @@ import './Login.scss';
 
-        <Login_Button />
+        <Login_Button
+          idfunc={this.state.id}
+          pwfunc={this.state.pw}
+          colorfunc={
+            id.indexOf('@') !== -1 && pw.length >= 8 ? 'blue' : 'sky_blue'
+          }
```
idValue를 넘겨주는 걸로 보이는데 idfunc라는 이름은 마치 함수같아서 적합하지 않아보입니다!
pwfunc, colorfunc 모두 마찬가지입니다!
그리고 colorfunc라는 값은 id랑 pw통해서 계산 가능해보이는데 그러면 idValue랑 pwValue만 Props로 받고, colorfunc에서 계산 하는 값은 LoginButton 컴포넌트에서 props통해서 바로 계산해서 사용해도 무방해보입니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -1,14 +1,62 @@
+  }
+
+  callApi = () => {
+    fetch(this.props.fetch, {
```
api주소가 data안에 들어있는 형태는 비정상적인 것 같습니다!
api주소는 일반적으로 백엔드에서 전송해주지 않습니다!
comment정보랑 feed정보를 분리시켜놓으셔서 그런 것 같은데, feed안에 comment정보를 그냥 다 넣어주시면 될 것 같습니다!

```
  {
    id: 1,
    profilesrc: '/images/jaesangChoi/profile1.jpeg',
    mainfeedsrc:
      'https://images.unsplash.com/photo-1623659993428-0c5504a16ca5?ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw0Mnx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60',
  comment:["foo","bar"]
  },
```
이런식으로요! 코멘트란 정보는 피드에 종속되어 있는 정보니까 저런 형태가 적합해보입니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -1,14 +1,62 @@
+
+  callApi = () => {
+    fetch(this.props.fetch, {
+      method: 'GET',
```
get method는 기본값이라서 따로 기입안해주셔도 됩니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -1,14 +1,62 @@
+
+    this.setState({
+      comments: this.state.comments.concat({
+        id: this.state.comments.length + 1,
```
댓글은 나중에 삭제될 수도 있으니까 id를 length로 사용하면 삭제 후 다시 추가되는 경우 동일한 id를 가진 댓글이 생길 것 같습니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -1,14 +1,62 @@
+      comment: '',
+    });
+
+    return;
```
함수는 return문 따로 없으면 끝까지 다 읽고나서 알아서 종료되기때문에 임의로 return 키워드를 적을 필요는 없어보입니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -1,14 +1,62 @@
           <div className="feeds_profile">
             <div className="feeds_profile_image">
-              <img alt="프로필" src="/images/jaesangChoi/profile1.jpeg" />
+              <img alt="프로필" src={this.props.profilesrc} />
```
여기는 alt를 한글로 적고 다른 곳은 영어로 적으신 이유가 있으실까요?? 영어로 적어주시는게 좋아보입니다!
한글이 더 좋지만,, 만국 공용어가 영어기 때문에 보편성을 위해서 😂

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -73,19 +121,27 @@ class Feeds extends Component {
           <div className="third_comment">42분 전</div>
 
+          <ul className="comment_list">
+            {comments.map(t => {
```
t라는 변수명을 가지고는 여기에 할당된 값이 뭐일지 전혀 유추가 안되는 것 같습니다!
comments 배열안의 하나하나의 요소들이니까 comment라고 이름지어주는게 좋지 않을까요?
```suggestion
            {comments.map(comment => {
```

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/Feeds.js:
```diff
@@ -73,19 +121,27 @@ class Feeds extends Component {
               <img alt="스마일아이콘" src="/images/jaesangChoi/smile.png" />
             </div>
-            <div className="push_comment_text">
+            <form className="push_comment_text" onSubmit={pressButton}>
```
마찬가지로 pressButton이라는 이름만 보고는 전혀 유추가 안됩니다!
pressButton이라는건 발생하는 시점이지 함수가 하고있는 동작이 아니기 때문에 함수의 역할을 동사형으로 명확하게 addComment 등으로 이름지어주는게 좋아보입니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Feeds/FeedsData.js:
```diff
@@ -0,0 +1,19 @@
+const FeedsData = [
```
상수데이터는 대문자 + snakecase 로 이름짓는 컨벤션이 있습니다!
지금은 파스칼케이스를 사용하고 계시네요
FEEDS_DATA로 변경해주세요!

yeonuk-hwang replied:
다른 상수데이터도 모두 마찬가지입니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main.js:
```diff
@@ -2,16 +2,41 @@ import React, { Component } from 'react';
         <div className="main">
-          <Feeds />
+          <div>
+            {feedslist.map(flist => {
```
feedslist안의 하나의 요소들인데 flist라는 변수명은 적합하지 않은것 같습니다! 차라리 feed등이 좀 더 적합하고 의미가 드러나 보이네요!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main.js:
```diff
@@ -2,16 +2,41 @@ import React, { Component } from 'react';
       <div>
         <Nav />
-
         <div className="main">
```
메인 콘텐츠 구획을 의미하는 거라면 div tag보다는 main tag가 좀 더 적합하지 않을까요?

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main.scss:
```diff
@@ -1,31 +1,18 @@
+  display: flex;
+  justify-content: center;
+  gap: 20px;
+  width: 100%;
```
이미 display:block인 요소인데 width:100% 를 부여할 이유가 없어보입니다!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main.scss:
```diff
@@ -1,31 +1,18 @@
+  justify-content: center;
+  gap: 20px;
+  width: 100%;
+  height: 100vh;
```
height:100vh로 부모태그의 크기를 고정값으로 정하는건 좋지않습니다!
부모태그에 고정높이값을 주는건 스크롤이 필요한 경우를 제외하고는 지양하고있습니다!
안에 여러 자식태그들이 들어올텐데 그 크기만큼 자연스럽게 늘어나도록 height값을 주지 않는게 좋아보입니다!

Higher77 replied:
main.scss에서 main태그에 height: 100vh를 준 이유가 화면 전체를 범위를 갖게 해주려고 사용했습니다.
그 이유는 Main_right의 박스가 Nav처럼 고정된걸 구현하려고 했고 그 방법으로 position:sticky가 떠올랐기 때문입니다.
sticky로 화면에 계속 고정시키기 위해선 화면 전체에 대해서 범위를 가져야하는데 Main_right는 main태그의 자식이기 때문에
main태그가 화면전체의 범위를 가져야 했습니다.

그래서 main에 height:100vh를 줬습니다아... height를 auto로 설정하면 position:sticky가 정상적으로 작동하지 않습니다 ㅠㅠ;

Higher77 replied:
라운지에 계실 때 여쭤보길 잘한거 같습니다. 
덕분에 position:sticky에 대해서 착각했던게 완전히 잡혔습니다.
더이상 position은 두렵지 않네요!!!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main_right/Main_right.js:
```diff
@@ -1,9 +1,25 @@
 import React, { Component } from 'react';
 import Recommendation_list from './Recommendation_list/Recommendation_list';
+import RecommendationData from './RecommendationData';
```
상수데이터는 대문자 + snake_case 사용해주세요!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main_right/Main_right.js:
```diff
@@ -1,9 +1,25 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      Recommendationlist: [],
```
camelCase 사용해주세요! 지금은 pascalCase를 사용하고 계시네요!

yeonuk-hwang commented on src/pages/jaesangChoi/Main/Main_right/Main_right.js:
```diff
@@ -26,29 +42,18 @@ class Main_right extends Component {
+                <Recommendation_list
+                  key={list.id}
+                  src={list.src}
+                  main_name={list.main_name}
```
camelCase 사용해주세요!

yeonuk-hwang commented on public/data/JaesangChoi/commentDataJaesang.json:
```diff
@@ -0,0 +1,14 @@
+  },
+  {
+    "id": 3,
+    "text": "앞으로 잘부탁드립니다"
```
잘 부탁드립니다🙌

yeonuk-hwang reviewed (commented):
수고하셨습니다 재상님!
리뷰 확인 & 반영해주세요!

Commit:
[Refactor] idfunc,pwfunc,colorfunc 이름 수정 및 Props 수정

Commit:
[Refactor] Mock데이터 구조 변경
-commentJaesang.json, commentjaesang2.json파일을 FeedsData.js로 통합
-FeedsDate.js파일의 확장자를 json으로 변환
-Feed안에 쓰인 fetch를 Main으로 끌어올림.
-Feed에 componentDidMount를 추가하여 FeedsData의 comments를 setState함.

Commit:
[Refactor] 변수, 함수 및 데이터를 컨벤션에 따라 수정, 어울리는 네이밍으로 변경

Commit:
[Refactor] 잘못 사용된 main.scss height값 삭제
